platform/view/services/comm/p2p_test.go

```diff
@@ -105,7 +113,16 @@ func setupTwoNodesFromFiles(t *testing.T) (*P2PNode, *P2PNode, string, string) {
 	bootstrapNodeEndpoint := "/ip4/127.0.0.1/tcp/1234"
 	nodeEndpoint := "/ip4/127.0.0.1/tcp/1235"
 
-	bootstrapNode, node := setupTwoNodes(t, bootstrapNodeID, bootstrapNodeEndpoint, nodeID, nodeEndpoint, &PrivateKeyFromFile{bootstrapNodeSK}, &PrivateKeyFromFile{nodeSK})
+	// sometimes setupTwoNodes can fail due to timeouts, give it a second chance
+	var err error
+	var bootstrapNode, node *P2PNode
+	for i := 0; i < 3; i++ {
+		bootstrapNode, node, err = setupTwoNodes(t, bootstrapNodeID, bootstrapNodeEndpoint, nodeID, nodeEndpoint, &PrivateKeyFromFile{bootstrapNodeSK}, &PrivateKeyFromFile{nodeSK})
+		if err == nil {
+			break
+		}
+	}
```

Comment:
I suggest a dedicated Retry function, which can be used in many places.

This is cleaner than just running the code in the loop adding a comment that we want to retry ... 

For instance:
```go
func retry(attempts int, sleep time.Duration, f func() error) (err error) {
	for i := 0; i < attempts; i++ {
		if i > 0 {
			time.Sleep(sleep)
			sleep *= 2
		}

		err = f()
		if err == nil {
			return nil
		}
	}
	return fmt.Errorf("no luck after %d attempts: last error: %v", attempts, err)
}
```

```go
var bootstrapNode, node *P2PNode
retry(3, 0 *time.Seconds, func() error {
  bootstrapNode, node, err = setupTwoNodes(t, bootstrapNodeID, bootstrapNodeEndpoint, nodeID, nodeEndpoint, &PrivateKeyFromFile{bootstrapNodeSK}, &PrivateKeyFromFile{nodeSK})
  if err != nil {
    return err 
  }
})
```

Reply:
beautiful, let me do that

Reply:
done